Sources/Jetworking/Client/Interceptor/Protocol/Interceptor.swift

```diff
@@ -1,4 +1,43 @@
 import Foundation
 
-/// Base Protocol for `RequestInterceptor` and `ResponseInterceptor`.
-public protocol Interceptor {}
+/// Base Protocol for `Interceptor`.
+/// `Interceptors` are able to intercept a request or a response.
+public protocol Interceptor {
+    /**
+     * # Summary
+     * Intercepting the request.
+     *
+     * - Parameter request:
+     *  The request to be intercepted.
+     *
+     * - Returns:
+     * The intercepted request.
+     */
+    func intercept(_ request: URLRequest) -> URLRequest
```

Comment:
Is this function definition explanatory enough? I mean in this case we are intercepting the request but below the response. One can inherit it from the documentation and the parameters but somehow I am uncertain if this is explanatory enough.